src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNamespace.cs

```diff
@@ -16,14 +16,14 @@ public sealed class XNamespace
         internal const string xmlPrefixNamespace = "http://www.w3.org/XML/1998/namespace";
         internal const string xmlnsPrefixNamespace = "http://www.w3.org/2000/xmlns/";
 
-        private static XHashtable<WeakReference> s_namespaces;
-        private static WeakReference s_refNone;
-        private static WeakReference s_refXml;
-        private static WeakReference s_refXmlns;
+        private static XHashtable<WeakReference<XNamespace>> s_namespaces;
+        private static WeakReference<XNamespace> s_refNone;
+        private static WeakReference<XNamespace> s_refXml;
+        private static WeakReference<XNamespace> s_refXmlns;
 
         private readonly string _namespaceName;
         private readonly int _hashCode;
-        private readonly XHashtable<XName> _names;
+        private readonly XHashtable<WeakReference<XName>> _names;
```

Comment:
What are those numbers comparing?

Reply:
As perf results was stable i had measured before change and after change 3 times each and compared their combinations. Now I think better to setup Linux env and run the tests

Reply:
> Did you experiment with using a ConditionalWeakTable instead of XHashtable?

@stephentoub I tried but that is not really suitable for this issue.